### PR TITLE
Drishti package (some fixes)

### PR DESCRIPTION
**Pull request by ruslo**

### CODE REVIEW

File: CMakeLists.txt [hunk 1]
```diff
@@ -17,8 +18,17 @@ list(APPEND CMAKE_MODULE_PATH "${CMAKE_CURRENT_LIST_DIR}/drishti-upload/cmake/Mo
 # drishti specific hunter cache pacakges:
 list(APPEND HUNTER_CACHE_SERVERS "https://github.com/elucideye/hunter-cache")
 
-include(drishti_set_hunter_gate)
-drishti_set_hunter_version(DRISHTI_HUNTER_GATE_URL DRISHTI_HUNTER_GATE_SHA1)
+if(EXISTS "${drishti_upload_modules}")
+  include(drishti_set_hunter_gate)
+  drishti_set_hunter_version(DRISHTI_HUNTER_GATE_URL DRISHTI_HUNTER_GATE_SHA1)
+else()
+  # Release archive will not contain submodule 'drishti-upload'.
```
Comment (ruslo) [inline CMakeLists.txt]: Workaround for https://github.com/headupinclouds/drishti_hunter_test/issues/1

File: cmake/Modules/drishti_split_debug_symbols.cmake [hunk 2]
```diff
@@ -19,18 +19,23 @@ function(drishti_split_debug_symbols lib_name)
   else()
 
     # see: https://sourceware.org/gdb/onlinedocs/gdb/Separate-Debug-Files.html
-    set(BUILDID "abcdef1234")
-    string(SUBSTRING "${BUILDID}" 0 2 BUILDIDPREFIX)
-    string(SUBSTRING "${BUILDID}" 2 8 BUILDIDSUFFIX)
+    string(SHA1 build_id "${PROJECT_NAME} ${lib_name}")
```
Comment (headupinclouds) [inline cmake/Modules/drishti_split_debug_symbols.cmake]: A definite improvement :)

File: cmake/Modules/drishti_split_debug_symbols.cmake [hunk 3]
```diff
@@ -27,11 +27,11 @@ function(drishti_split_debug_symbols lib_name)
     drishti_check_cxx_linker_flag("-Wl,--build-id=0x${BUILDID}" linker_supports_build_id)
 
     message("linker_supports_build_id: ${linker_supports_build_id}")
-    
+
```
Comment (headupinclouds) [inline cmake/Modules/drishti_split_debug_symbols.cmake]: I'm still waiting for cmake-format.

File: CMakeLists.txt (hunk 1 above)
Comment (headupinclouds) [reply CMakeLists.txt]: 👍

**Comment by headupinclouds** [comment]
Thanks.

**Pull request merged**